### ISSUE TITLE
System.NotSupportedException: 'Unable to activate instance of type Xamarin.Forms.PancakeView.Droid.PancakeViewRenderer from native handle 0x100019 (key_handle 0x213296cd).'

### ISSUE THREAD

**Issue by LeoJHarris** [sthewissen/Xamarin.Forms.PancakeView#116]
Error thowing in latest nuget 2.0.1 when I navigate back to a navigation stack that has the pancake view and type into an entry with a parent pancakeview wrapping around it., I am also using FreshMVVM. I tested under the same conditions and found version 1.4.2 to work fine.

Bassically doing something like this:

```csharp
            CoreMethods.SwitchOutRootNavigation(NavigationStacks.MainAppStack);
// Later in code I go back to the Login stack and crash occurs attempting to get back to that nav stack
            CoreMethods.SwitchOutRootNavigation(NavigationStacks.LoginNavigationStack);
```

I.e. Navigation Stack A is just contentpage with pancakeview, then swap to navigation stack B and swap back to navigation stack A again.

```csharp
07-21 15:21:34.841 D/TextView(16653): setTypeface with style : 0
07-21 15:21:34.912 E/ViewRootImpl(16653): sendUserActionEvent() mView == null
07-21 15:21:35.862 D/ViewRootImpl(16653): ViewPostImeInputStage ACTION_DOWN
07-21 15:21:37.734 D/Mono    (16653): Requesting loading reference 7 (of 8) of Mono.Android.dll
07-21 15:21:37.734 D/Mono    (16653): Loading reference 7 of Mono.Android.dll asmctx DEFAULT, looking for System.Runtime.Serialization, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e
07-21 15:21:37.734 D/Mono    (16653): Assembly Ref addref Mono.Android[0x9e2a16a0] -> System.Runtime.Serialization[0x9bc1d7c0]: 3
**System.NotSupportedException:** 'Unable to activate instance of type Xamarin.Forms.PancakeView.Droid.PancakeViewRenderer from native handle 0x100019 (key_handle 0x213296cd).'

Thread finished: <Thread Pool> #5
The thread 0x5 has exited with code 0 (0x0).
Thread finished: <Thread Pool> #10
The thread 0xa has exited with code 0 (0x0).
07-21 15:21:48.245 D/Mono    (16653): DllImport attempting to load: '/system/lib/liblog.so'.Thread started: <Thread Pool> #19

07-21 15:21:48.255 D/Mono    (16653): DllImport loaded library '/system/lib/liblog.so'.
07-21 15:21:48.255 D/Mono    (16653): DllImport searching in: '/system/lib/liblog.so' ('/system/lib/liblog.so').
07-21 15:21:48.255 D/Mono    (16653): Searching for '__android_log_print'.
07-21 15:21:48.255 D/Mono    (16653): Probing '__android_log_print'.
07-21 15:21:48.255 D/Mono    (16653): Found as '__android_log_print'.
07-21 15:21:48.295 I/MonoDroid(16653): UNHANDLED EXCEPTION:
07-21 15:21:48.295 I/MonoDroid(16653): System.NotSupportedException: Unable to activate instance of type Xamarin.Forms.PancakeView.Droid.PancakeViewRenderer from native handle 0x100019 (key_handle 0x213296cd). ---> System.MissingMethodException: No constructor found for Xamarin.Forms.PancakeView.Droid.PancakeViewRenderer::.ctor(System.IntPtr, Android.Runtime.JniHandleOwnership) ---> Java.Interop.JavaLocationException: Exception of type 'Java.Interop.JavaLocationException' was thrown.
07-21 15:21:48.295 I/MonoDroid(16653):    --- End of inner exception stack trace ---
07-21 15:21:48.295 I/MonoDroid(16653):   at Java.Interop.TypeManager.CreateProxy (System.Type type, System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x000b5] in <eaa205f580954a64824b74a79fa87c62>:0 
07-21 15:21:48.295 I/MonoDroid(16653):   at Java.Interop.TypeManager.CreateInstance (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type targetType) [0x00111] in <eaa205f580954a64824b74a79fa87c62>:0 
07-21 15:21:48.295 I/MonoDroid(16653):    --- End of inner exception stack trace ---
07-21 15:21:48.305 I/MonoDroid(16653):   at Java.Interop.TypeManager.CreateInstance (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type targetType) [0x0017e] in <eaa205f580954a64824b74a79fa87c62>:0 
07-21 15:21:48.305 I/MonoDroid(16653):   at Java.Lang.Object.GetObject (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type type) [0x00023] in <eaa205f580954a64824b74a79fa87c62>:0 
07-21 15:21:48.305 I/MonoDroid(16653):   at Java.Lang.Object._GetObject[T] (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x00017] in <eaa205f580954a64824b74a79fa87c62>:0 
07-21 15:21:48.305 I/MonoDroid(16653):   at Java.Lang.Object.GetObject[T] (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x00000] in <eaa205f580954a64824b74a79fa87c62>:0 
07-21 15:21:48.305 I/MonoDroid(16653):   at Android.Views.View.get_Parent () [0x0001f] in <eaa205f580954a64824b74a79fa87c62>:0 
07-21 15:21:48.305 I/MonoDroid(16653):   at Xamarin.Forms.Platform.Android.DescendantFocusToggler.RequestFocus (Android.Views.View control, System.Func`1[TResult] baseRequestFocus) [0x0002f] in D:\a\1\s\Xamarin.Forms.Platform.Android\Renderers\DescendantFocusToggler.cs:27 
07-21 15:21:48.305 I/MonoDroid(16653):   at Xamarin.Forms.Material.Android.MaterialFormsEditTextBase.Xamarin.Forms.Platform.Android.IDescendantFocusToggler.RequestFocus (Android.Views.View control, System.Func`1[TResult] baseRequestFocus) [0x00015] in D:\a\1\s\Xamarin.Forms.Material.Android\MaterialFormsEditTextBase.cs:51 
07-21 15:21:48.305 I/MonoDroid(16653):   at Xamarin.Forms.Material.Android.MaterialFormsEditTextBase.RequestFocus (Android.Views.FocusSearchDirection direction, Android.Graphics.Rect previouslyFocusedRect) [0x0001b] in D:\a\1\s\Xamarin.Forms.Material.Android\MaterialFormsEditTextBase.cs:56 
07-21 15:21:48.305 I/MonoDroid(16653):   at Android.Views.View.n_RequestFocus_ILandroid_graphics_Rect_ (System.IntPtr jnienv, System.IntPtr native__this, System.Int32 native_direction, System.IntPtr native_previouslyFocusedRect) [0x00011] in <eaa205f580954a64824b74a79fa87c62>:0 
07-21 15:21:48.305 I/MonoDroid(16653):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.84(intptr,intptr,int,intptr)
07-21 15:21:48.325 E/AppCenterCrashes(16653): Unhandled Exception from source=AndroidEnvironment
07-21 15:21:48.325 E/AppCenterCrashes(16653): System.NotSupportedException: Unable to activate instance of type Xamarin.Forms.PancakeView.Droid.PancakeViewRenderer from native handle 0x100019 (key_handle 0x213296cd). ---> System.MissingMethodException: No constructor found for Xamarin.Forms.PancakeView.Droid.PancakeViewRenderer::.ctor(System.IntPtr, Android.Runtime.JniHandleOwnership) ---> Java.Interop.JavaLocationException: Exception of type 'Java.Interop.JavaLocationException' was thrown.
07-21 15:21:48.325 E/AppCenterCrashes(16653):    --- End of inner exception stack trace ---
07-21 15:21:48.325 E/AppCenterCrashes(16653):   at Java.Interop.TypeManager.CreateProxy (System.Type type, System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x000b5] in <eaa205f580954a64824b74a79fa87c62>:0 
07-21 15:21:48.325 E/AppCenterCrashes(16653):   at Java.Interop.TypeManager.CreateInstance (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type targetType) [0x00111] in <eaa205f580954a64824b74a79fa87c62>:0 
07-21 15:21:48.325 E/AppCenterCrashes(16653):    --- End of inner exception stack trace ---
07-21 15:21:48.325 E/AppCenterCrashes(16653):   at Java.Interop.TypeManager.CreateInstance (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type targetType) [0x0017e] in <eaa205f580954a64824b74a79fa87c62>:0 
07-21 15:21:48.325 E/AppCenterCrashes(16653):   at Java.Lang.Object.GetObject (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type type) [0x00023] in <eaa205f580954a64824b74a79fa87c62>:0 
07-21 15:21:48.325 E/AppCenterCrashes(16653):   at Java.Lang.Object._GetObject[T] (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x00017] in <eaa205f580954a64824b74a79fa87c62>:0 
07-21 15:21:48.325 E/AppCenterCrashes(16653):   at Java.Lang.Object.GetObject[T] (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x00000] in <eaa205f580954a64824b74a79fa87c62>:0 
07-21 15:21:48.325 E/AppCenterCrashes(16653):   at Android.Views.View.get_Parent () [0x0001f] in <eaa205f580954a64824b74a79fa87c62>:0 
07-21 15:21:48.325 E/AppCenterCrashes(16653):   at Xamarin.Forms.Platform.Android.DescendantFocusToggler.RequestFocus (Android.Views.View control, System.Func`1[TResult] baseRequestFocus) [0x0002f] in D:\a\1\s\Xamarin.Forms.Platform.Android\Renderers\DescendantFocusToggler.cs:27 
07-21 15:21:48.325 E/AppCenterCrashes(16653):   at Xamarin.Forms.Material.Android.MaterialFormsEditTextBase.Xamarin.Forms.Platform.Android.IDescendantFocusToggler.RequestFocus (Android.Views.View control, System.Func`1[TResult] baseRequestFocus) [0x00015] in D:\a\1\s\Xamarin.Forms.Material.Android\MaterialFormsEditTextBase.cs:51 
07-21 15:21:48.325 E/AppCenterCrashes(16653):   at Xamarin.Forms.Material.Android.MaterialFormsEditTextBase.RequestFocus (Android.Views.FocusSearchDirection direction, Android.Graphics.Rect previouslyFocusedRect) [0x0001b] in D:\a\1\s\Xamarin.Forms.Material.Android\MaterialFormsEditTextBase.cs:56 
07-21 15:21:48.325 E/AppCenterCrashes(16653):   at Android.Views.View.n_RequestFocus_ILandroid_graphics_Rect_ (System.IntPtr jnienv, System.IntPtr native__this, System.Int32 native_direction, System.IntPtr native_previouslyFocusedRect) [0x00011] in <eaa205f580954a64824b74a79fa87c62>:0 
07-21 15:21:48.325 E/AppCenterCrashes(16653):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.84(intptr,intptr,int,intptr)
07-21 15:21:48.345 D/Mono    (16653): Requesting loading reference 5 (of 6) of /storage/emulated/0/Android/data/com.sample.MyApp/files/.__override__/Microsoft.AppCenter.Crashes.dll
07-21 15:21:48.345 D/Mono    (16653): Loading reference 5 of /storage/emulated/0/Android/data/com.sample.MyApp/files/.__override__/Microsoft.AppCenter.Crashes.dll asmctx DEFAULT, looking for System.Core, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e
07-21 15:21:48.345 D/Mono    (16653): Assembly Ref addref Microsoft.AppCenter.Crashes[0xb4784b20] -> System.Core[0x9e2a1fa0]: 15
07-21 15:21:48.495 D/Mono    (16653): DllImport searching in: '__Internal' ('(null)').
07-21 15:21:48.495 D/Mono    (16653): Searching for 'java_interop_jnienv_throw'.
07-21 15:21:48.495 D/Mono    (16653): Probing 'java_interop_jnienv_throw'.
07-21 15:21:48.495 D/Mono    (16653): Found as 'java_interop_jnienv_throw'.
[0:] Entry: Failed to set CursorPosition from renderer: System.NotSupportedException: Unable to activate instance of type Xamarin.Forms.PancakeView.Droid.PancakeViewRenderer from native handle 0x100019 (key_handle 0x213296cd). ---> System.MissingMethodException: No constructor found for Xamarin.Forms.PancakeView.Droid.PancakeViewRenderer::.ctor(System.IntPtr, Android.Runtime.JniHandleOwnership) ---> Java.Interop.JavaLocationException: Exception of type 'Java.Interop.JavaLocationException' was thrown.
   --- End of inner exception stack trace ---
  at Java.Interop.TypeManager.CreateProxy (System.Type type, System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x000b5] in <eaa205f580954a64824b74a79fa87c62>:0 
  at Java.Interop.TypeManager.CreateInstance (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type targetType) [0x00111] in <eaa205f580954a64824b74a79fa87c62>:0 
   --- End of inner exception stack trace ---
  at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.84(intptr,intptr,int,intptr)
  at (wrapper native-to-managed) Android.Runtime.DynamicMethodNameCounter.84(intptr,intptr,int,intptr)
  at (wrapper managed-to-native) Java.Interop.NativeMethods.java_interop_jnienv_call_nonvirtual_boolean_method_a(intptr,intptr&,intptr,intptr,intptr,intptr)
  at Java.Interop.JniEnvironment+InstanceMethods.CallNonvirtualBooleanMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniObjectReference type, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args) [0x00079] in <9324da45a6654f83baffa7c2854d836a>:0 
  at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeNonvirtualBooleanMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x0001f] in <9324da45a6654f83baffa7c2854d836a>:0 
  at Android.Views.View.RequestFocus () [0x0000a] in <eaa205f580954a64824b74a79fa87c62>:0 
  at Xamarin.Forms.Platform.Android.EntryRendererBase`1[TControl].UpdateCursorSelection () [0x00026] in D:\a\1\s\Xamarin.Forms.Platform.Android\Renderers\EntryRenderer.cs:457 
  at Xamarin.Forms.Platform.Android.EntryRendererBase`1[TControl].OnElementPropertyChanged (System.Object sender, System.ComponentModel.PropertyChangedEventArgs e) [0x00303] in D:\a\1\s\Xamarin.Forms.Platform.Android\Renderers\EntryRenderer.cs:295 
  at (wrapper delegate-invoke) <Module>.invoke_void_object_PropertyChangedEventArgs(object,System.ComponentModel.PropertyChangedEventArgs)
  at Xamarin.Forms.BindableObject.OnPropertyChanged (System.String propertyName) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:229 
  at Xamarin.Forms.Element.OnPropertyChanged (System.String propertyName) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Element.cs:353 
  at Xamarin.Forms.BindableObject.SetValueActual (Xamarin.Forms.BindableProperty property, Xamarin.Forms.BindableObject+BindablePropertyContext context, System.Object value, System.Boolean currentlyApplying, Xamarin.Forms.Internals.SetValueFlags attributes, System.Boolean silent) [0x00114] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:461 
  at Xamarin.Forms.BindableObject.SetValueCore (Xamarin.Forms.BindableProperty property, System.Object value, Xamarin.Forms.Internals.SetValueFlags attributes, Xamarin.Forms.BindableObject+SetValuePrivateFlags privateAttributes) [0x00173] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:397 
  at Xamarin.Forms.BindableObject.SetValueCore (Xamarin.Forms.BindableProperty property, System.Object value, Xamarin.Forms.Internals.SetValueFlags attributes) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:343 
  at Xamarin.Forms.Element.SetValueFromRenderer (Xamarin.Forms.BindableProperty property, System.Object value) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Element.cs:250 
  at Xamarin.Forms.Element.Xamarin.Forms.IElementController.SetValueFromRenderer (Xamarin.Forms.BindableProperty property, System.Object value) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Element.cs:246 
  at Xamarin.Forms.Platform.Android.EntryRendererBase`1[TControl].SetCursorPositionFromRenderer (System.Int32 start) [0x00007] in D:\a\1\s\Xamarin.Forms.Platform.Android\Renderers\EntryRenderer.cs:511 

=================================================================
07-21 15:21:48.515 F/libc    (16653): Fatal signal 11 (SIGSEGV), code 1, fault addr 0x120020a0 in tid 16653 (uite.MyApp)	Native Crash Reporting
=================================================================
Got a SEGV while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

No native Android stacktrace (see debuggerd output).

=================================================================
	Basic Fault Address Reporting
=================================================================
Memory around native instruction pointer (0xb4ea8250):0xb4ea8240  2e 82 11 b0 bd e8 f0 8f fa 69 53 1c f1 d0 bd 68  .........iS....h
0xb4ea8250  ee 6b 75 03 ed d4 df f8 08 88 39 46 04 22 54 f8  .ku.......9F."T.
0xb4ea8260  08 30 1d 68 05 f5 f8 70 80 f7 98 fa 80 46 df f8  .0.h...p.....F..
0xb4ea8270  f4 07 24 58 a0 45 dc d0 d5 f8 d4 20 d2 f8 28 61  ..$X.E..... ..(a

=================================================================
	Managed Stacktrace:
=========================================

========================
	  at <unknown> <0xffffffff>
	  at Java.Interop.NativeMethods:java_interop_jnienv_call_nonvirtual_object_method_a <0x00057>
	  at InstanceMethods:CallNonvirtualObjectMethod <0x00367>
	  at JniInstanceMethods:InvokeVirtualObjectMethod <0x00397>
	  at Android.Widget.TextView:get_TextFormatted <0x000df>
	  at Android.Widget.TextView:get_Text <0x0005b>
	  at Xamarin.Forms.Platform.Android.EntryRendererBase`1:SelectionChanged <0x00297>
	  at Xamarin.Forms.Material.Android.MaterialFormsEditText:OnSelectionChanged <0x00117>
	  at Android.Widget.TextView:n_OnSelectionChanged_II <0x000af>
	  at Android.Runtime.DynamicMethodNameCounter:86 <0x0013b>
	  at Android.Runtime.DynamicMethodNameCounter:86 <0x000f3>
=================================================================
```

**Comment by sthewissen** [comment]
Similar to #113

**Comment by LeoJHarris** [comment]
@sthewissen opps didnt realize that ticket was open, yes I think the same issue.

**Comment by sthewissen** [comment]
Could you try this in https://www.nuget.org/packages/Xamarin.Forms.PancakeView/2.0.1.701-beta?

**Comment by ilpork** [comment]
I had similar issue when using Shell navigation. First I navigated to page 1, then to page 2, then to page 3, then selected page 1 again and tried to enter text in an `Entry` wrapped in `PancakeView`. It crashed but with the 2.0.1.701-beta it seems to work fine.

**Comment by LeoJHarris** [comment]
@sthewissen Updated nuget to preview and launch app but crashed before it even got the page:

System.MissingMethodException: 'Constructor on type 'Xamarin.Forms.PancakeView.Droid.PancakeViewRenderer' not found.'

**Comment by sthewissen** [comment]
Is a Visual = Material applied implicitly/explicitly? Pretty sure that is the problem.

**Comment by LeoJHarris** [comment]
@sthewissen yes Visual was applied explicitly.

```csharp
/// <summary>
        /// Initializes a new instance of the <see cref="BasePopupPage"/> class.
        /// </summary>
        public BasePopupPage()
        {
            if (Device.RuntimePlatform == Device.Android)
            {
                Visual = VisualMarker.Material;
            }

            On<Xamarin.Forms.PlatformConfiguration.iOS>().SetUseSafeArea(true);
        }
```

2.0.1.701-beta resolved all this original issue with material turned off.

**Comment by sthewissen** [comment]
Could you try the newest beta (707)? I tried setting Visual to Default in PancakeView to try and break that inheritance as it has no Visual implementation anyway.

**Comment by LeoJHarris** [comment]
@sthewissen Seems like it's working now in newest beta (707).

App can launch with visual applied explicitly and navigate back again without crashing.

**Comment by jackbond** [comment]
I found this issue because I was getting a MissingMethodException with the PancakeView. I'm using v2.1.0.714. I was able to make the exception go away by setting the Visual to Default. I'm working on some code that I have inherited, so I actually have no idea what property does. That said, it doesn't seem like it should be raising an exception. Will this be fixed in the 2.3 / vNext?

**Comment by sthewissen** [comment]
There's a 2.3 beta out, could you try that for this?
